src/components/container/ProviderSearch/ProviderSearch.tsx

```diff
@@ -99,10 +99,12 @@ export default function (props: ProviderSearchProps) {
     function connectToProvider(provider: ExternalAccountProvider) {
         const providerID = provider.id;
         if (!props.previewState && !(linkedExternalAccounts[providerID] && linkedExternalAccounts[providerID].status != 'unauthorized')) {
-            MyDataHelps.connectExternalAccount(providerID, { openNewWindow: props.openNewWindow ?? false });
-        }
-        if (props.onProviderSelected) {
-            props.onProviderSelected(provider);
+            MyDataHelps.connectExternalAccount(providerID, { openNewWindow: props.openNewWindow ?? false })
+                .then(function() {
```

Comment:
Is this promise rejected when the user cancels out of the flow or has an error?  Are we sure this only captures "successful connection" scenarios?

Reply:
Might try to write a simple WebViewStep in MDH that calls ```MyDataHelps.connectExternalAccount``` to test the various results of the promise across iOS/Android/Web to ensure we have this working 100% and does not allow proceeding if the user cancels, for instance

Reply:
> Is this promise rejected when the user cancels out of the flow or has an error? Are we sure this only captures "successful connection" scenarios?

Correct. I've tested it in Long COVID QA. The promise only gets resolved after a user completes all the authorization steps at the 3rd party website. When a participant backs out, e.g., click "Cancel" at Garmin, they will stay at the same survey step.

Reply:
https://github.com/CareEvolution/MyDataHelpsUI/pull/116#discussion_r1333546831